src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -375,7 +389,7 @@ private void OutputRelatedType(NodeFactory factory, ref ObjectDataBuilder objDat
             {
                 relatedType = ((ParameterizedType)_type).ParameterType;
             }
-            
+
             if (relatedType != null)
             {
                 if (_constructed)
```

Comment:
Unrelated to this change - for arrays and pointers, it should be always NecessaryTypeSymbol (e.g. creating array of valuetype that is never boxed should not cause the full vtable generated to be generated for the valuetype).